### PR TITLE
maintainer: Use build_cmake script in icp cuda builds.

### PR DESCRIPTION
**Pull request by fweik**
Fixes #

Description of changes:
 - 

PR Checklist
------------
 - [ ] Tests?
   - [ ] Interface
   - [ ] Core 
 - [ ] Docs?

### CODE REVIEW

**Comment by fweik** [comment]
Coverage is not yet working, this needs adding the codecov token to the secrets in gitlab, which I lack permission for.

File: src/python/espressomd/lb.pyx [hunk 1]
```diff
@@ -40,7 +40,6 @@ IF LB_GPU or LB:
     cdef class LBFluid(HydrodynamicInteraction):
         """
         Initialize the lattice-Boltzmann method for hydrodynamic flow using the CPU.
-
```
Comment (KaiSzuttor) [inline src/python/espressomd/lb.pyx:43]: these empty lines are part of the numpydoc style recommondations

File: src/python/espressomd/lb.pyx [hunk 2]
```diff
@@ -209,7 +208,6 @@ IF LB_GPU:
     cdef class LBFluid_GPU(LBFluid):
         """
         Initialize the lattice-Boltzmann method for hydrodynamic flow using the GPU.
-
```
Comment (KaiSzuttor) [inline src/python/espressomd/lb.pyx:212]: same here

File: src/core/communication.cpp [hunk 3]
```diff
@@ -2076,12 +2076,6 @@ void mpi_bcast_lb_params(int field, int value) {
 }
 
 void mpi_bcast_lb_params_slave(int field, int value) {
-#if defined(LB) || defined(LB_GPU)
```
Comment (KaiSzuttor) [inline src/core/communication.cpp:2079]: are these changes intended to be part of this PR?

File: src/core/unit_tests/ParallelScriptInterface_test.cpp [hunk 4]
```diff
@@ -96,35 +83,37 @@ BOOST_AUTO_TEST_CASE(ctor_dtor) {
   }
 
   /* Check that ctor and dtor were run on all nodes */
-  Testing::reduce_and_check(callbacks->comm(), TestClass::constructed);
```
Comment (KaiSzuttor) [inline src/core/unit_tests/ParallelScriptInterface_test.cpp:99]: Are these changes intended to be part of this PR?

**Review by KaiSzuttor — CHANGES_REQUESTED** [review]
~~You can also delete the old build file:
https://github.com/espressomd/espresso/blob/8673af53e19202ed05e1f50010f21c7fa57eff75/maintainer/cuda_build.sh~~

Overall looks good to me, let's wait for the builds after the secret is placed into the gitlab project

**Comment by KaiSzuttor** [comment]
You now have permission to add a secret variable to the gitlab project

File: src/core/unit_tests/ParallelScriptInterface_test.cpp (hunk 4 above)
Comment (fweik) [reply src/core/unit_tests/ParallelScriptInterface_test.cpp:99]: Yes, I turned out that the test was broken.

**Comment by KaiSzuttor** [comment]
why is ICC failing? There is also a problem with the artifacts but that seems to be a problem of the gitlab server

**Comment by fweik** [comment]
The problem with the artifacts is that they are now too big, because I switched from default to maxset.

**Comment by fweik** [comment]
ICC fails some times because of the P3M tuning. The test should use appropriate fixed P3M parameters, ~~will fix.~~ Fixed in #1409.

**Comment by fweik** [comment]
Apparently the max artifact size has to be configured globaly (c.f. https://gitlab.com/gitlab-org/gitlab-ce/issues/20612)

**Comment by KaiSzuttor** [comment]
I just set it to 1 GB, just tested your branch and the build dir has approx. 400 MB

**Comment by fweik** [comment]
Jup, works (https://gitlab.icp.uni-stuttgart.de/fweik/espresso/-/jobs/595)

**Comment by KaiSzuttor** [comment]
lcov was missing in the docker image, just PR'ed to docker repo

**Review by KaiSzuttor — APPROVED** [review]

File: maintainer/cuda_build.sh [hunk 5]
```diff
@@ -9,11 +9,7 @@ curl "https://api.github.com/repos/espressomd/espresso/statuses/$GIT_COMMIT?acce
       -d "{\"state\": \"$STATUS\", \"context\": \"ICP CUDA build\", \"target_url\": \"$URL\"}"
 
 nvidia-smi
-cd ${CI_PROJECT_DIR}
-cp maintainer/configs/maxset.hpp myconfig.hpp
-mkdir build && cd build
-cmake .. -DCUDA_NVCC_FLAGS="-D_FORCE_INLINES"
-make -j 4 && make check_python
+srcdir="${CI_PROJECT_DIR}" cmake_params='-DCUDA_NVCC_FLAGS="-D_FORCE_INLINES"' with_coverage="true" build_procs=4 maintainer/travis/build_cmake.sh
```
Comment (mkuron) [inline maintainer/cuda_build.sh]: I think `-DCUDA_NVCC_FLAGS="-D_FORCE_INLINES` can be removed. We now have that in CMakeLists.txt for NVCC version 8 and GCC version 5.

**Review by mkuron — APPROVED** [review]

**Comment by fweik** [comment]
Concurrent builds seem to work too...
